lib.rs

```diff
@@ -89,14 +107,23 @@ impl<D: Decorator> Format<D> {
 
         try!(self.print_msg_header(io, &r_decorator, record));
         let mut serializer = Serializer::new(io, r_decorator);
+        let mut skip_msg_comma = NullWriter.write_fmt(record.msg()).is_ok();
```

Comment:
The `CountingWriter` idea is great - I like that a lot better than what I've implemented. I'll make the changes and update the PR.